wgpu-core/src/device/resource.rs

```diff
@@ -690,7 +735,11 @@ impl Device {
         self: &Arc<Self>,
         hal_buffer: Box<dyn hal::DynBuffer>,
         desc: &resource::BufferDescriptor,
-    ) -> Arc<Buffer> {
+    ) -> Result<Arc<Buffer>, resource::CreateBufferError> {
+        #[cfg(feature = "indirect-validation")]
+        let raw_indirect_validation_bind_group =
+            self.create_indirect_validation_bind_group(hal_buffer.as_ref(), desc.size, desc.usage)?;
+
         unsafe { self.raw().add_raw_buffer(&*hal_buffer) };
```

Comment:
@nical, was it correct to resolve a conflict here by ordering the `add_raw_buffer` counter increment after creating `raw_indirect_validation_bind_group`?

Reply:
The patch looks correct to me: if `create_indirect_validation_bind_group` fails then the `Buffer` is not added and `destroy_buffer` will not be called so the counter is balanced.